google/cloud/spanner/client.h

```diff
@@ -0,0 +1,193 @@
+#ifndef GOOGLE_CLOUD_CPP_GOOGLE_CLOUD_SPANNER_CLIENT_H_
+#define GOOGLE_CLOUD_CPP_GOOGLE_CLOUD_SPANNER_CLIENT_H_
+
+#include "google/cloud/spanner/key.h"
+#include "google/cloud/spanner/mutation.h"
+#include "google/cloud/spanner/partition.h"
+#include "google/cloud/spanner/result_set.h"
+#include "google/cloud/spanner/result_stats.h"
+#include "google/cloud/spanner/row.h"
+#include "google/cloud/spanner/sql_statement.h"
+#include "google/cloud/spanner/transaction.h"
+#include "google/cloud/spanner/value.h"
+#include "google/cloud/spanner/version.h"
+#include "google/cloud/status.h"
+#include "google/cloud/status_or.h"
+#include <chrono>
+#include <string>
+#include <vector>
+
+namespace google {
+namespace cloud {
+namespace spanner {
+inline namespace SPANNER_CLIENT_NS {
+
+class Client {
+ public:
+  // move-only
+  Client(Client&&) = default;
+  Client& operator=(Client&&) = default;
+  Client(Client const&) = delete;
+  Client& operator=(Client const&) = delete;
+
+  //
+  // Read()
+  //
+
+  // Reads the columns for the given keys from the specified table. Returns a
+  // stream of Rows.
+  // TODO: Add support for "limit" integer.
+  ResultSet Read(Transaction tx, std::string table, KeySet keys,
+                 std::vector<std::string> columns) {
+    // Fills in two rows of dummy data.
+    std::vector<Row> v;
+    int64_t data = 1;
+    v.push_back(Row{});
+    for (auto const& c : columns) {
+      (void)c;  // unused
+      v.back().AddValue(Value(data++));
+    }
+    v.push_back(Row{});
+    for (auto const& c : columns) {
+      (void)c;  // unused
+      v.back().AddValue(Value(data++));
+    }
+    return ResultSet(v);
+  }
+
+  // Same as Read() above, but implicitly uses a single-use Transaction.
+  ResultSet Read(std::string table, KeySet keys,
+                 std::vector<std::string> columns) {
+    auto single_use = Transaction::MakeSingleUseTransaction();
+    return Read(std::move(single_use), std::move(table), std::move(keys),
+                std::move(columns));
+  }
+
+  ResultSet Read(ReadPartition partition) {
+    // TODO: Call Spanner's StreamingRead RPC with the data in `partition`.
```

Comment:
These TODO's are sort of implicit, but they should also refer to bugs, maybe create one bug that says "convert all the TODOs in client.h to bugs and link them?"